lib/document/DocumentFactory.js

```diff
@@ -240,44 +240,60 @@ class DocumentFactory {
           throw new InvalidInitialRevisionError(document);
         }
 
-        const json = {
+        const plainObject = document.toObject();
+
+        const keysToStay = [
+          '$id',
+          '$type',
+          '$dataContractId',
+          '$createdAt',
+          '$updatedAt',
+        ];
+
+        Object.keys(plainObject).forEach((key) => {
+          if (key.startsWith('$') && !keysToStay.includes(key)) {
+            delete plainObject[key];
+          }
+        });
+
+        return {
+          ...plainObject,
           $action: AbstractDocumentTransition.ACTIONS.CREATE,
-          $id: document.getId(),
-          $type: document.getType(),
-          $dataContractId: document.getDataContractId(),
           $entropy: document.getEntropy(),
-          ...document.getData(),
         };
-
-        if (document.getCreatedAt()) {
-          json.$createdAt = document.getCreatedAt().getTime();
```

Comment:
@jawid-h Maybe raw data should contain Date objects and only for JSON, we convert them to integers, what do you think?

Reply:
Probably later :)